desktop-widgets/divelogexportdialog.cpp

```diff
@@ -442,26 +401,25 @@ void DiveLogExportDialog::export_TeX(const char *filename, const bool selected_o
 		qty_weight = 0;
 		total_weight = 0;
 		for (i = 0; i < MAX_WEIGHTSYSTEMS; i++){
-			if (dive->weightsystem[i].weight.grams != NULL){
+			if (dive->weightsystem[i].weight.grams){
 				put_format(&buf, "\\def\\weight%dtype{%s}\n", i + 1, dive->weightsystem[i].description);
 				put_format(&buf, "\\def\\weight%damt{%.3f \\weightunit}\n", i + 1, get_weight_units(dive->weightsystem[i].weight.grams, NULL, &unit));
 				qty_weight += 1;
 				total_weight += get_weight_units(dive->weightsystem[i].weight.grams, NULL, &unit);
+			} else {
+				put_format(&buf, "\\def\\weight%dtype{}\n", i + 1);
+				put_format(&buf, "\\def\\weight%damt{}\n", i + 1);
 			}
 		}
 		put_format(&buf, "\\def\\qtyweights{%d}\n", qty_weight);
 		put_format(&buf, "\\def\\totalweight{%.2f \\weightunit}\n", total_weight);
 		unit = "";
 
-		// Deprecated fields
-		put_format(&buf, "\n%% Deprecated Fields - kept to maintain compatibility with legacy templates:\n");
-		put_format(&buf, "%% The following fields are deprecated and may not return any value!\n");
-		put_format(&buf, "%% They have been let here to preserve functionality in legacy templates.");
-		put_format(&buf, "%% Do not use these fields in future templates.\n");
-		put_format(&buf, "\\def\\spot{} %% Deprecated - does not return a value\n"); // Deprecated - remains for use in legacy templates
-		put_format(&buf, "\\def\\entrance{} %% Deprecated - does not return a value\n"); // Deprecated - remains for use in legacy templates
-		put_format(&buf, "\\def\\place{%s} %% Deprecated - use \\sitename instead\n", site ? site->name : ""); // Deprecated - remains for use in legacy templates
-		get_depth_units(dive->maxdepth.mm, NULL, &unit) > 0 ? put_format(&buf, "\\def\\depth{%.1f \\depthunit}%% Deprecated - use \\maximumdepth instead\n", get_depth_units(dive->maxdepth.mm, NULL, &unit)) : put_format(&buf, "\\def\\depth{} %% Deprecated - use \\maximumdepth instead\n");  // Deprecated - remains for use in legacy templates
+		// Legacy fields
+		put_format(&buf, "\\def\\spot{}\n");
```

Comment:
Writing this as a person who doesn't know much, I think it would be good to leave the note about deprecated fields here - it serves two purposes: first, it keeps people from designing new templates with these fields. Second it tells idiots like me who are monkeying with the code that they should not be casually deleted (someone will be upset)!